test/fixtures/hdfs-fixture/src/main/java/org/elasticsearch/test/fixtures/hdfs/HdfsClientThreadLeakFilter.java

```diff
@@ -29,6 +29,11 @@ public final class HdfsClientThreadLeakFilter implements ThreadFilter {
 
     @Override
     public boolean reject(Thread t) {
-        return t.getName().equals(OFFENDING_THREAD_NAME);
+        return t.getName().contains(OFFENDING_THREAD_NAME)
+            || t.getName().startsWith("LeaseRenewer")
+            || t.getName().startsWith("SSL Certificates Store Monitor") // hadoop 3 brings that in
+            || t.getName().startsWith("GcTimeMonitor") // hadoop 3
+            || t.getName().startsWith("Command processor") // hadoop 3
+            || t.getName().startsWith("ForkJoinPool-"); // hadoop 3
```

Comment:
"ForkJoinPool" is pretty generic. Annoying we can't be more specific here.

Reply:
@mark-vieira One option could be that we test by default against hdfs2 (I haven't seen forkedjoinedpool there) and then only explicity filter this out explicitly for hdfs3. It's not the nicest solution but that way we reduce the surface of that filtering. WDYT?

Reply:
I think if we're only applying this filter to HDFS tests then it's probably not worth the extra effort.